v2/contribute/decisions/oauth/0007-use-jwt-access-tokens.mdx

```diff
@@ -0,0 +1,45 @@
+---
```

Comment:
Some other ADRs:
- Do we need to store the access token hash in the db at all? Right now, we store it only to be able to provide an endpoint wherein the user can validate if the access token exists in the db. But this can also be done by checking against the `id` in the access token, which is already in the db (similar to the concept of `sessionHandle` for session recipe). The advantage of this is that, if we provide an endpoint to change the access token claims (is that even a thing?) then older access tokens will continue to work even from a db checking point of view - since the `id` will not change.
- Related to the above point, should we provide an endpoint to update the access token claims? What if the user's roles change, then devs may want new access and id tokens for that user right? How do other auth providers handle this?
- We will need to add the access token and id token payload in the db cause we allow users to give custom claims in them, and post refresh, we need to be able to re add those claims in the new tokens. Any better approach other than storing the claims in the db?

Reply:
Also, which code challenge methods should we support - for example, just SHA256 or more?

Reply:
> The advantage of this is that, if we provide an endpoint to change the access token claims (is that even a thing?) then older access tokens will continue to work even from a db checking point of view - since the id will not change.

The token claims could change over time, basically if you call refresh your information in the token should be updated. 
Changing the payload for the current access token has it's issues:
- we should only issue a new access token during refresh calls. 
- Since the ability to change the "current" token is missing I don't think we should add an endpoint to do it "offline".
- You can not add new scopes during refresh, which kind of means that you should not add any new information

Common use-cases, like requiring MFA can be done by redirecting the user to the auth server with additional required scopes. Keep in mind, that this may not require the user to re-enter their password, they can be just redirected to the MFA screen.

Older access tokens working isn't an advantage if you can't change them outside of the refresh endpoint. It'd basically mean that old tokens will keep working even if checking the DB directly after you called refresh.

> Any better approach other than storing the claims in the db?

I've outlined this elsewhere but let's discuss this on a call.